Makefile

```diff
@@ -2,91 +2,66 @@ SHELL = /usr/bin/env bash
-SCRIPTS_PATH = $(shell pwd)/build/scripts
 SRC_PATH = $(shell pwd)/src
-TOOLSET_SRC_PATH = $(shell pwd)/build/MSBuildToolset
-TOOLSET_PATH = $(BINARIES_PATH)/toolset
-RESTORE_SEMAPHORE_PATH = $(BINARIES_PATH)/restore.semaphore
 BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
 BUILD_LOG_PATH =
 HOME_DIR = $(shell cd ~ && pwd)
-DOTNET_VERSION = 1.0.0-preview3-003223
-NUGET_VERSION = 3.5.0-beta2
-NUGET_EXE = $(shell pwd)/nuget.exe
+DOTNET_VERSION = 1.0.1
+DOTNET = $(BINARIES_PATH)/dotnet-cli/dotnet
+TARGET_FX = netcoreapp1.1
 
 MSBUILD_ADDITIONALARGS := /v:m /fl /fileloggerparameters:Verbosity=normal /p:Configuration=$(BUILD_CONFIGURATION)
 
 ifeq ($(OS_NAME),Linux)
-	MSBUILD_ADDITIONALARGS := $(MSBUILD_ADDITIONALARGS) /p:BaseNuGetRuntimeIdentifier=$(shell . /etc/os-release && echo $$ID.$$VERSION_ID)
+	RUNTIME_ID := $(shell . /etc/os-release && echo $$ID.$$VERSION_ID)-x64
 else ifeq ($(OS_NAME),Darwin)
-	MSBUILD_ADDITIONALARGS := $(MSBUILD_ADDITIONALARGS) /p:BaseNuGetRuntimeIdentifier=osx.10.12
+	RUNTIME_ID := osx.10.12-x64
 endif
 
 ifneq ($(BUILD_LOG_PATH),)
 	MSBUILD_ADDITIONALARGS := $(MSBUILD_ADDITIONALARGS) /fileloggerparameters:LogFile=$(BUILD_LOG_PATH)
 endif
 
 ifeq ($(BOOTSTRAP),true)
-	MSBUILD_ARGS = $(MSBUILD_ADDITIONALARGS) /p:CscToolPath=$(BOOTSTRAP_PATH) /p:CscToolExe=csc /p:VbcToolPath=$(BOOTSTRAP_PATH) /p:VbcToolExe=vbc
+	MSBUILD_ARGS = $(MSBUILD_ADDITIONALARGS) /p:CscToolPath=$(BOOTSTRAP_PATH)/csc /p:CscToolExe=csc /p:VbcToolPath=$(BOOTSTRAP_PATH)/vbc /p:VbcToolExe=vbc
 else
-	MSBUILD_ARGS = $(MSBUILD_ADDITIONALARGS) /p:CscToolExe=csc /p:VbcToolExe=vbc
+	MSBUILD_ARGS = $(MSBUILD_ADDITIONALARGS)
 endif
 
-MSBUILD_CMD = $(TOOLSET_PATH)/corerun $(TOOLSET_PATH)/MSBuild.dll $(MSBUILD_ARGS)
+BUILD_CMD = dotnet build $(MSBUILD_ARGS)
 
-.PHONY: all bootstrap test restore toolset nuget
+.PHONY: all bootstrap test toolset
 
-all: $(RESTORE_SEMAPHORE_PATH)
-	export HOME=$(HOME_DIR) ; \
-	$(MSBUILD_CMD) CrossPlatform.sln
+all: restore
+	@export PATH="$(BINARIES_PATH)/dotnet-cli:$(PATH)" ; \
+	$(BUILD_CMD) CrossPlatform.sln
 
-bootstrap: $(TOOLSET_PATH) $(RESTORE_SEMAPHORE_PATH)
-	export HOME=$(HOME_DIR) ; \
-	$(MSBUILD_CMD) src/Compilers/CSharp/CscCore/CscCore.csproj && \
-	$(MSBUILD_CMD) src/Compilers/VisualBasic/VbcCore/VbcCore.csproj && \
-	mkdir -p $(BOOTSTRAP_PATH) && \
-	cp -f Binaries/$(BUILD_CONFIGURATION)/Exes/CscCore/* $(BOOTSTRAP_PATH) && \
-	cp -f Binaries/$(BUILD_CONFIGURATION)/Exes/VbcCore/* $(BOOTSTRAP_PATH)
-
-ifneq ($(SKIP_CROSSGEN),true)
-	build/scripts/crossgen.sh $(BOOTSTRAP_PATH)
-endif
+bootstrap: restore
+	export PATH="$(BINARIES_PATH)/dotnet-cli:$(PATH)" ; \
+	$(BUILD_CMD) src/Compilers/CSharp/CscCore && \
+	$(BUILD_CMD) src/Compilers/VisualBasic/VbcCore && \
+	mkdir -p $(BOOTSTRAP_PATH)/csc && mkdir -p $(BOOTSTRAP_PATH)/vbc && \
+	dotnet publish -r $(RUNTIME_ID) src/Compilers/CSharp/CscCore -o $(BOOTSTRAP_PATH)/csc && \
+	dotnet publish -r $(RUNTIME_ID) src/Compilers/VisualBasic/VbcCore -o $(BOOTSTRAP_PATH)/vbc
 
```

Comment:
There is an errant blank line here.  Is that intentional?  Not crisp on makefile rules but this seems wrong at a glance.